plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java

```diff
@@ -27,327 +27,289 @@
-      LOG.error("Unable to fetch database node structure");
-      throw e;
+
+    Structure getEdgeStructure(NodeVersion nodeVersion) throws GroundException {
+        try {
+            LOG.debug("Fetching database table edge structure: " + nodeVersion.getNodeId());
+            Edge edge = this.getEdge(nodeVersion);
+            return groundReadWrite.getStructure(edge.getName());
+        } catch (GroundException e) {
+            LOG.debug("Not found - database table edge structure: " + nodeVersion.getNodeId());
+            throw new GroundException(e);
+        }
     }
-  }
-
-  Edge getEdge(NodeVersion nodeVersion) throws GroundException {
-    String edgeId = "" + nodeVersion.getNodeId();
-    try {
-      LOG.debug("Fetching database table edge: " + edgeId);
-      return groundReadWrite.getEdgeFactory().retrieveFromDatabase(edgeId);
-    } catch (GroundException e) {
-      LOG.debug("Not found - Creating database table edge: " + edgeId);
-      Edge edge = groundReadWrite.getEdgeFactory().create(edgeId);
-      Structure edgeStruct = groundReadWrite.getStructureFactory().create(edge.getName());
-      return edge;
+
+    Database getDatabase(String dbName) throws GroundException {
+        List<String> versions = groundReadWrite.getLatestVersions(dbName);
+        if (versions.isEmpty()) {
+            throw new GroundException("Database node not found: " + dbName);
+        }
+        NodeVersion latestVersion = groundReadWrite.getNodeVersion(versions.get(0));
+        Map<String, Tag> dbTag = latestVersion.getTags();
+        return JsonUtil.fromJSON((String) dbTag.get(dbName).getValue(), Database.class);
     }
-  }
-
-  Structure getEdgeStructure(NodeVersion nodeVersion) throws GroundException {
-    try {
-      LOG.debug("Fetching database table edge structure: " + nodeVersion.getNodeId());
-      Edge edge = this.getEdge(nodeVersion);
-      return groundReadWrite.getStructureFactory().retrieveFromDatabase(edge.getName());
-    } catch (GroundException e) {
-      LOG.debug("Not found - database table edge structure: " + nodeVersion.getNodeId());
-      throw e;
+
+    NodeVersion createDatabase(Database db) throws InvalidObjectException, MetaException {
+        if (db == null) {
+            throw new InvalidObjectException("Database object passed is null");
+        }
+        try {
+            String dbName = db.getName();
+            Node dbNode = this.getNode(dbName);
+            Structure dbStruct = groundReadWrite.createStructure(dbName);
+            LOG.debug("Node and Structure {}, {}", dbNode.getId(), dbStruct.getId());
+            Map<String, GroundType> structVersionAttribs = new HashMap<>();
+            structVersionAttribs.put(dbName, GroundType.STRING);
+            StructureVersion sv = groundReadWrite.createStructureVersion(dbStruct.getName(), dbStruct.getId(),
+                    structVersionAttribs);
+
+            Tag dbTag = new Tag("1.0.0", dbName, JsonUtil.toJSON(db), GroundType.STRING);
```

Comment:
I pulled upstream master in to fork and changed these to long - how are we going to generate these uniqueIds - just a sequence of versions?